src/acl_hal_mmd.cpp

```diff
@@ -2767,10 +2767,19 @@ void *acl_hal_mmd_host_alloc(const std::vector<cl_device_id> devices,
     handles[i] = device_info[physical_device_id].handle;
   }
 
-  result = dispatch->aocl_mmd_host_alloc(
-      handles, devices.size(), size, alignment,
-      (aocl_mmd_mem_properties_t *)properties, error);
-  acl_delete_arr(handles);
+  int use_offline_only;
+  acl_get_offline_device_user_setting(&use_offline_only);
+
+  if (use_offline_only == ACL_CONTEXT_MPSIM) {
+    result = dispatch->aocl_mmd_host_alloc(
+        handles, devices.size(), size, alignment,
+        (aocl_mmd_mem_properties_t *)properties, error);
```

Comment:
You may be able to reduce the amount of common code in the `if-else` by simply modifying `properties`, or using a sacrificial pointer that you set to `properties` if we're in sim mode?